### PR TITLE
Xilinx platform mangement related changes

### PR DESCRIPTION
**Pull request by sivadur**
This series contains xilinx platform/power management related changes which are fixes/ minor feature additions.

### CODE REVIEW

File: plat/xilinx/zynqmp/aarch64/zynqmp_common.c [hunk 1]
```diff
@@ -59,6 +61,7 @@ static const struct {
 	unsigned int id;
 	unsigned int ver;
 	char *name;
+	bool evexists;
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/aarch64/zynqmp_common.c]: nit: It's a bit weird to use a `bool` if you are going to assign to it integer values (1 or 0).

Comment (sivadur) [reply plat/xilinx/zynqmp/aarch64/zynqmp_common.c]: Thanks for your comment, will use true or false instead of 1 or 0.

**Comment by sivadur** [comment]
@antonio-nino-diaz-arm 

Updated with your comment .
Also, i added a new patch to the series for handling wdt reset incase of wdt timeout in the system.

File: plat/xilinx/zynqmp/pm_service/pm_svc_main.c [hunk 2]
```diff
@@ -32,6 +45,144 @@ static struct {
...
+
+	spin_lock(&inc_lock);
+	active_cores--;
+
+	for (i = 0; i < 4; i++) {
+		mmio_write_32(BASE_GICD_BASE + GICD_CPENDSGIR + 4 * i,
+				0xffffffff);
+	}
+
+	dsb();
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/pm_service/pm_svc_main.c]: This `dsb` isn't needed, the store release inside `spin_unlock` already enforces the previous writes to be seen before the write that unlocks the spinlock.

Check "K10.2.1 Message passing" in the Arm ARM (version B.b). The example "Resolving weakly-ordered message passing by using Acquire and Release" is a very simple one and explains this.

**Comment by antonio-nino-diaz-arm** [comment]
This PR needs a rebase.

**Comment by sivadur** [comment]
@antonio-nino-diaz-arm  Sure, I will fix review comments and will rebase while doing it.

**Pull request closed without merging**